src/app_service/service/network/network_item.nim

```diff
@@ -0,0 +1,214 @@
+import NimQml, stew/shims/strformat
+
+import ./dto
+
+QtObject:
+  type NetworkItem* = ref object of QObject
+    chainId*: int
+    layer*: int
+    chainName*: string
+    iconUrl*: string
+    shortName*: string
+    chainColor*: string
+    rpcURL*: string
+    fallbackURL*: string
+    originalRpcURL*: string
+    originalFallbackURL*: string
+    blockExplorerURL*: string
+    nativeCurrencyName*: string
+    nativeCurrencySymbol*: string
+    nativeCurrencyDecimals*: int
+    isTest*: bool
+    isEnabled*: bool
+    enabledState*: UxEnabledState
+    relatedChainId*: int
+
+  proc setup*(self: NetworkItem,
+    chainId: int,
+    layer: int,
+    chainName: string,
+    iconUrl: string,
+    shortName: string,
+    chainColor: string,
+    rpcURL: string,
+    fallbackURL: string,
+    originalRpcURL: string,
+    originalFallbackURL: string,
+    blockExplorerURL: string,
+    nativeCurrencyName: string,
+    nativeCurrencySymbol: string,
+    nativeCurrencyDecimals: int,
+    isTest: bool,
+    isEnabled: bool,
+    enabledState: UxEnabledState,
+    relatedChainId: int
+    ) =
+      self.QObject.setup
+      self.chainId = chainId
+      self.layer = layer
+      self.chainName = chainName
+      self.iconUrl = iconUrl
+      self.shortName = shortName
+      self.chainColor = chainColor
+      self.rpcURL = rpcURL
+      self.fallbackURL = fallbackURL
+      self.originalRpcURL = originalRpcURL
+      self.originalFallbackURL = originalFallbackURL
+      self.blockExplorerURL = blockExplorerURL
+      self.nativeCurrencyName = nativeCurrencyName
+      self.nativeCurrencySymbol = nativeCurrencySymbol
+      self.nativeCurrencyDecimals = nativeCurrencyDecimals
+      self.isTest = isTest
+      self.isEnabled = isEnabled
+      self.enabledState = enabledState
+      self.relatedChainId = relatedChainId
+
+  proc delete*(self: NetworkItem) =
+      self.QObject.delete
+
+  proc networkDtoToItem*(network: NetworkDto): NetworkItem =
+    new(result, delete)
+    result.setup(network.chainId, network.layer, network.chainName, network.iconUrl, network.shortName,
+      network.chainColor, network.rpcURL, network.fallbackURL, network.originalRpcURL, network.originalFallbackURL,
+      network.blockExplorerURL, network.nativeCurrencyName, network.nativeCurrencySymbol, network.nativeCurrencyDecimals,
+      network.isTest, network.enabled, network.enabledState, network.relatedChainId)
+
+  proc `$`*(self: NetworkItem): string =
+    result = fmt"""NetworkItem(
+      chainId: {self.chainId},
+      chainName: {self.chainName},
+      layer: {self.layer},
+      iconUrl:{self.iconUrl},
+      shortName: {self.shortName},
+      chainColor: {self.chainColor},
+      rpcURL: {self.rpcURL},
+      fallbackURL: {self.fallbackURL},
+      blockExplorerURL: {self.blockExplorerURL},
+      nativeCurrencySymbol: {self.nativeCurrencySymbol},
+      ]"""
+
+  proc chainIdChanged*(self: NetworkItem) {.signal.}
```

Comment:
@Khushboo-dev-cpp you can remove all these signals since they are not used at all, exposed properties are just `CONSTANT`.